port/platform/zephyr/CMakeLists.txt

```diff
@@ -78,17 +78,13 @@ if (CONFIG_UBXLIB_TEST)
         ${UBXLIB_TEST_INC}
     )
 
-# If we are building outside of Platform IO then we
-# bring in unity from the NRF Connect SDK in which
-# Zephyr is contained.  However, under Platform IO that
-# doesn't exist, you get it from Platform IO
-# instead, hence make this a "weak" inclusion
-if (EXISTS "${ZEPHYR_BASE}/../test/cmock/vendor/unity/src/unity.c")
+# west clones unity for us, but it's posible that west update hasn't been run yet
+if (EXISTS "${ZEPHYR_BASE}/../modules/test/cmock/vendor/unity")
```

Comment:
I'll see how the test system takes it: thanks for much for this, will try to sort it tomorrow (Wednesday).

Reply:
Because the test system is designed to work with various SDKs (ESP-IDF, nRF5SDK, STM32Cube etc.) it has its own Python scripts which ensure that known versions of all the tools it needs for a given job are installed.  In the Zephyr case it ensures that nRFConnectSDK (v2.3.0), GCC for ARM and make are installed and, after that, it calls `west` to do the build.

In terms of where `unity` comes from, the test system does have a copy of its own (2.5.0) but it only uses that for the SDKs that don't already have an associated `unity` version; where the SDK has a version of `unity` we always try to use that version, just 'cos that's going to be closest to what the customer would do, which is how we ended up using the nRFConnectSDK one.

That said, it would be better, more general, to use a version of `unity` that is linked with Zephyr; however I can't find a `.west` directory in any of the builds, I guess because we've, so far, not done this the `manifest` way.  Which means that I guess we'd need to add that.

Reply:
We could just specify unity 2.5.0 in the ubxlib west.yml and have the test system bring it's own version in.
That way a person using ubxlib gets unity 2.5.0, and the CI stuff also has it.

Reply:
Sounds good.